examples/MicroserviceExample/WebApi/RabbitMqService.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using Microsoft.Extensions.Logging;
+using OpenTelemetry.Context.Propagation;
+using RabbitMQ.Client;
+
+namespace WebApi
+{
+    public interface IRabbitMqService : IDisposable
+    {
+        string PublishMessage();
+    }
+
+    public class RabbitMqService : IRabbitMqService
+    {
+        private const string QueueName = "TestQueue";
+
+        private readonly ILogger<RabbitMqService> logger;
+        private readonly ConnectionFactory connectionFactory;
+        private readonly IConnection connection;
+        private readonly IModel channel;
+        private readonly ActivitySource activitySource;
+        private readonly ITextFormat textFormat;
+
+        public RabbitMqService(ILogger<RabbitMqService> logger)
+        {
+            this.logger = logger;
+
+            this.connectionFactory = new ConnectionFactory()
+            {
+                HostName = Environment.GetEnvironmentVariable("RABBIT_HOSTNAME") ?? "localhost",
+                UserName = "guest",
+                Password = "guest",
+                Port = 5672,
+                RequestedConnectionTimeout = TimeSpan.FromMilliseconds(3000),
+            };
+
+            this.connection = connectionFactory.CreateConnection();
+            this.channel = connection.CreateModel();
+            channel.QueueDeclare(
+                queue: QueueName,
+                durable: false,
+                exclusive: false,
+                autoDelete: false,
+                arguments: null);
+
+            this.activitySource = new ActivitySource(nameof(RabbitMqService));
+
+            this.textFormat = new TraceContextFormat();
+        }
+
+        public string PublishMessage()
+        {
+            try
+            {
+                string activityName = $"{nameof(RabbitMqService)}.{nameof(PublishMessage)}";
```

Comment:
[Spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/messaging.md#span-name) says operationName should be `<destination name> <operation name>` example `$"{QueueName} send"` for publish here. I just noticed that yesterday it might be new?

Reply:
Ah cool, great call out. I'm really glad to see there's guidance here.